web/controllers/user_controller.ex

```diff
@@ -41,6 +41,16 @@ defmodule Pairmotron.UserController do
     user = Repo.get!(User, id)
     changeset = User.changeset(user, user_params)
 
+    if current_assigned_user_id(conn) != id && changeset.changes[:password] do
```

Comment:
I am going to leave this for a separate PR, but you are correct. This should be refactored.